cmd/analyzetestresults/analyzetestresults.go

```diff
@@ -0,0 +1,221 @@
+	switch f.failureType {
+	case OtherFailure:
+		return
+	case ClusterCreationFailure:
+		testCaseEntry := returnContentWrappedInDropdown(dropdownSummaryString, f.clusterProvisionLog)
+		f.failedTestCaseNames = append(f.failedTestCaseNames, testCaseEntry)
+		return
+	case TestRunFailure:
+		testCaseEntry := returnContentWrappedInDropdown(dropdownSummaryString, f.e2eTestLog)
+		f.failedTestCaseNames = append(f.failedTestCaseNames, testCaseEntry)
+		return
+	}
+	for _, testSuite := range overallJUnitSuites.TestSuites {
+		if testSuite.Failures > 0 || testSuite.Errors > 0 {
+			var tcMessage string
+			for _, tc := range testSuite.TestCases {
+				if tc.Failure != nil || tc.Error != nil {
+					klog.Infof("Found a Test Case (suiteName/testCaseName): %s/%s, that didn't pass", testSuite.Name, tc.Name)
+					switch {
+					case tc.Status == "timedout":
+						tcMessage = returnContentWrappedInDropdown(dropdownSummaryString, tc.SystemErr)
+					case tc.Failure != nil:
+						tcMessage = "```\n" + tc.Failure.Message + "\n```"
+					default:
+						tcMessage = "```\n" + tc.Error.Message + "\n```"
+					}
+
+					testCaseEntry := "* :arrow_right: " + "[**`" + tc.Status + "`**] " + tc.Name + "\n" + tcMessage
+					f.failedTestCaseNames = append(f.failedTestCaseNames, testCaseEntry)
+				}
+			}
+		}
+	}
+}
+
+// getFormattedReport returns the full report as a string
+func (f *FailedTestCasesReport) getFormattedReport() string {
+	msg := f.headerString
+	for _, failedTCName := range f.failedTestCaseNames {
+		msg += fmt.Sprintf("\n %s\n", failedTCName)
+	}
+
+	return msg
+}
+
+func returnContentWrappedInDropdown(summary, content string) string {
+	return "<details><summary>" + summary + "</summary><br><pre>" + content + "</pre></details>"
+}
+
+// AnalyzeTestResultsCmd represents the analyze-test-results command
+var AnalyzeTestResultsCmd = &cobra.Command{
+	Use:   "analyze-test-results",
+	Short: "Command for analyzing test results",
+	PreRunE: func(cmd *cobra.Command, _ []string) error {
+		if viper.GetString(types.OciArtifactRefParamName) == "" {
+			_ = cmd.Usage()
+			return fmt.Errorf("parameter %q not provided, neither %s env var was set", types.OciArtifactRefParamName, types.OciArtifactRefEnv)
+		}
+		return nil
+	},
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		ociArtifactRef = viper.GetString(types.OciArtifactRefParamName)
+
+		cfg := oci.ScannerConfig{
+			OciArtifactReference: ociArtifactRef,
+			FileNameFilter:       []string{types.JunitFilename, clusterProvisionLogFilename, e2eTestRunLogFilename},
+		}
+
+		scanner, err := oci.NewArtifactScanner(cfg)
+		if err != nil {
+			return fmt.Errorf("failed to initialize artifact scanner: %+v", err)
+		}
+
+		if err := scanner.Run(); err != nil {
+			return fmt.Errorf("failed to scan artifact from %s: %+v", ociArtifactRef, err)
+		}
+
+		overallJUnitSuites, err := getTestSuitesFromXMLFile(scanner, jUnitFilename)
```

Comment:
I would assign the outputAllJunitSuites and let the Report struct own an instance of that